hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FSLeafQueue.java

```diff
@@ -557,28 +599,33 @@ private boolean preemptContainerPreCheck() {
         getFairShare());
   }
 
-  /**
-   * Is a queue being starved for its min share.
-   */
-  @VisibleForTesting
-  boolean isStarvedForMinShare() {
-    return isStarved(getMinShare());
+  private Resource minShareStarvation() {
+    Resource desiredShare = Resources.min(policy.getResourceCalculator(),
+        scheduler.getClusterResource(), getMinShare(), getDemand());
+
+    Resource starvation = Resources.subtract(desiredShare, getResourceUsage());
+    boolean starved = Resources.greaterThan(policy.getResourceCalculator(),
+        scheduler.getClusterResource(), starvation, none());
+
+    long now = scheduler.getClock().getTime();
+    if (!starved) {
+      setLastTimeAtMinShare(now);
+    }
+
+    if (starved &&
+        (now - lastTimeAtMinShare > getMinSharePreemptionTimeout())) {
+      return starvation;
+    } else {
+      return Resources.clone(Resources.none());
```

Comment:
You can make it:

~~~~
    if (!starved ||
        (now - lastTimeAtMinShare < getMinSharePreemptionTimeout())) {
      starvation = Resources.clone(Resources.none());

      if (!starved) {
        setLastTimeAtMinShare(now);
      }
    }

    return starvation;
~~~~